selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/hotupdate/RecreateUpdate.java

```diff
@@ -0,0 +1,171 @@
+  @Inject private CheTerminal terminal;
+  @Inject private Menu menu;
+
+  private int cheDeploymentBeforeRollout;
+
+  private enum WsMasterStauses {
+    RUNNING,
+    READY_TO_SHUTDOWN,
+    PREPARING_TO_SHUTDOWN,
+    SUSPENDED;
+  }
+
+  @BeforeClass
+  public void setUp() throws IOException {
+
+    cheDeploymentBeforeRollout =
+        parseInt(openShiftCliCommandExecutor.execute(COMMAND_FOR_GETTING_CURRENT_DEPLOYMENT_CHE));
+  }
+
+  @Test
+  public void checkRecreateUpdateStrategy() throws Exception {
+    String ocClientRolloutCommand = "rollout latest che";
+    String restUrlForSuspendingWorkspaces =
+        cheTestApiEndpointUrlProvider.get().toString() + "system/stop";
+    int timeLimitInSecForRecreateingUpdate = 600;
+    int delayBetweenRequest = 6;
+
+    // open a user workspace and send request for preparing to shutdown
+    ide.open(workspace);
+
+    testUserHttpJsonRequestFactory
+        .fromUrl(restUrlForSuspendingWorkspaces)
+        .usePostMethod()
+        .request();
+
+    // make sure, that system is prepared to  shutdown and than ready to shutdown
+    checkExpectedStatusesAndWorkspaceAfterShutDowning();
+
+    // performs rollout
+    openShiftCliCommandExecutor.execute(ocClientRolloutCommand);
+    waitOpenShiftStatus(
+        timeLimitInSecForRecreateingUpdate, delayBetweenRequest, WsMasterStauses.RUNNING);
+
+    // get current version of deployment after rollout
+    int cheDeploymentAfterRollout =
+        parseInt(openShiftCliCommandExecutor.execute(COMMAND_FOR_GETTING_CURRENT_DEPLOYMENT_CHE));
+    // After rollout updating - deployment should be increased on 1. So we previews version +1
+    // should be equal current
+    assertEquals(cheDeploymentAfterRollout, cheDeploymentBeforeRollout += 1);
+
+    // make sure that CHE ide is available after updating again
+    ide.open(workspace);
+    ide.waitOpenedWorkspaceIsReadyToUse();
+  }
+
+  private void checkExpectedStatusesAndWorkspaceAfterShutDowning() throws Exception {
+    int timeLimitForReadyToShutdownStatus = 30;
+    int delayBetweenRequestes = 1;
+
+    waitOpenShiftStatus(
+        timeLimitForReadyToShutdownStatus,
+        delayBetweenRequestes,
+        WsMasterStauses.PREPARING_TO_SHUTDOWN);
+
+    waitOpenShiftStatus(
+        timeLimitForReadyToShutdownStatus,
+        delayBetweenRequestes,
+        WsMasterStauses.READY_TO_SHUTDOWN);
+
+    // reopen the workspace and make sure that this one is not available after suspending system
+    ide.open(workspace);
+    projectExplorer.waitProjectExplorerIsNotPresent(timeLimitForReadyToShutdownStatus);
+    terminal.waitTerminalIsNotPresent(timeLimitForReadyToShutdownStatus);
+  }
+
+  private void waitOpenShiftStatus(
+      int maxTimeLimitInSec, int delayBetweenRequestsInSec, WsMasterStauses status)
+      throws Exception {
+
+    // if the limit is not exceeded - do request and check status of the system
+    while (maxTimeLimitInSec > 0) {
```

Comment:
It could be simplified as follow:
```
private void waitWorkspaceMasterStatus(
+      int maxReadStatusAttempts, int readStatusTimeoutInSec, WsMasterStatus expectedStatus)
+      throws Exception {
   int readStatusAttempts = maxReadStatusAttempts;
   while (readStatusAttempts-- > 0) {
      if (getCurrentRollingStatus().equals(expectedStatus.toString())) {
        return;
      }

      WaitUtils.sleepQuietly(readStatusTimeoutInSec);
   }

   throw new IOException(String.format("Workspace Master hasn't achieved status '%s' in '%' seconds.", status, maxReadStatusAttempts * readStatusTimeoutInSec));
   }
```